docker/entrypoint.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/sh
+
+set -o nounset
+set -o errexit
+set -o pipefail
+set -e
+
+restic="/usr/bin/restic"
+
+# if there are command arguments passed to the container execute them and exit.
+if [ $# -ne 0 ]; then
+  ${restic} $@
+  exit 0
+fi
+
+# check if the repo exists and initialize it if not
+${restic} cat config > /dev/null || ${restic} init
```

Comment:
The latest restic binaries return exit code 10 if a repositoy does not exist https://restic.readthedocs.io/en/latest/075_scripting.html#check-if-a-repository-is-already-initialized . It doesn't make sense to try to reinitialize the repository in other cases.

Besides that I'm not a fan of automatically initializing repositories. A repository should only be initialized once. If it disappears later on, there may be a major problem.